doc/classes/MainLoop.xml

```diff
@@ -1,10 +1,44 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="MainLoop" inherits="Object" category="Core" version="3.2">
 	<brief_description>
-		Main loop is the abstract main loop base class.
+		Abstract base class for the game's main loop.
 	</brief_description>
 	<description>
-		Main loop is the abstract main loop base class. All other main loop classes are derived from it. Upon application start, a [MainLoop] has to be provided to OS, else the application will exit. This happens automatically (and a [SceneTree] is created), unless a main [Script] is supplied, which may or not create and return a [MainLoop].
+		[MainLoop] is the abstract base class for a Godot project's game loop. It in inherited by [SceneTree], which is the default game loop implementation used in Godot projects, though it is also possible to write and use one's own [MainLoop] subclass instead of the scene tree.
+		Upon application start, a [MainLoop] implementation has to be provided to the OS, or the application will exit. This happens automatically (and a [SceneTree] is created) unless a main [Script] is provided from the command line (with e.g. [code]godot -s my_loop.gd[/code], which should then be a [MainLoop] implementation.
+		Here is an example script implementing a simple [MainLoop]:
+		[codeblock]
+		extends MainLoop
+
+		var time_elapsed = 0
+		var keys_typed = []
+		var quit = false
+
+		func _initialize():
+			print("Initialized:")
+			print("  Starting time: %s" % str(time_elapsed))
+
+		func _idle(delta):
+			time_elapsed += delta
+			# Return true to end the main loop
+			return quit
+
+		func _input_event(event):
+			# Record keys
+			if event is InputEventKey and event.pressed and !event.echo:
+				keys_typed.append(OS.get_scancode_string(event.scancode))
+				# Quit on Escape press
+				if event.scancode == KEY_ESCAPE:
+					quit = true
+			# Quit on any mouse click
+			if event is InputEventMouseButton:
+				quit = true
+
+		func _finalize():
+			print("Finalized:")
+			print("  End time: %s" % str(time_elapsed))
+			print("  Keys typed: %s" % var2str(keys_typed))
+		[/codeblock]
```

Comment:
I welcome suggestions for an actually useful example of extending MainLoop. I just hacked this together while documenting this class, but it would be nice to show how to render something simple too (currently it's just the boot logo blinking until the application closes).